src/agents.erl

```diff
@@ -93,3 +89,18 @@ torus_bounds(X, _Max) ->
 get_positions(Agents, Tile, Config) ->
     Impl = get_impl(Config),
     Impl:get_positions(Agents, Tile).
+
+-spec get_list(agents(), config()) -> [agent()].
+get_list(Agents, Config) ->
+    Impl = get_impl(Config),
+    Impl:get_list(Agents).
+
+-spec get_tiles(pos_integer(), environment(), config()) -> [{dimension(), tile()}].
+get_tiles(Dist, Env, Config) ->
+    Impl = get_impl(Config),
+    Impl:get_tiles(Dist, Env).
+
+-spec update_tiles({tile(), environment()}, environment(), config()) -> environment().
```

Comment:
Line 103 is too long: -spec update_tiles({tile(), environment()}, environment(), config()) -> environment()..